padcli/padcli.go

```diff
@@ -45,6 +48,11 @@ func New(opts ...padcliOption) *Padcli {
 	for _, opt := range opts {
 		opt(p)
 	}
+
+	if p.initSurverProvider == nil {
+		p.initSurverProvider = provider.DefaultInitSurveyProvider(p.clusterProvider)
```

Comment:
Do we need a separate if statement for this, instead of adding it straight to the Padcli struct above?

Reply:
This is ugly, but it was the best way I could think of that we always use the correct cluster provider (e.g. if the user of this lib  were to pass a custom cluster provider, but not a custom init provider, this ensures we assign it correctly).

If we do this above and there's a custom cluster provider then we will accidentally pass the default cluster provider.